packages/sdk/src/react/_internal/transaction-machine/execute-transaction.ts

```diff
@@ -591,6 +596,11 @@ export class TransactionMachine {
 				};
 
 				this.logger.debug('Opening payment modal', paymentModalProps);
+
+				if (changeCheckoutIsLoading) changeCheckoutIsLoading(false);
+
+				buyModal$.close();
```

Comment:
I have used something similar, and used external state in the machine which turns out multiple initializations of the machine.